### ISSUE TITLE
Make -x and --type mutually exclusive

### ISSUE THREAD

**Issue by petdance** [petdance/ack2#607]
The `--type` (and related `--perl`, `--php`, etc) flag has no effect when the `-x` flag is in use.  We should make ack warn about this.

Here's the use case:

I was searching for files that use Test::Deep but don't actually use the `cmp_bag` function.  So I used this:

    ack Test::Deep -l | ack -x -L cmp_bag

Works just fine, but it gave me hits on plain text files, so I ran it again as

    ack Test::Deep -l | ack -x -L cmp_bag --perl

The `--perl` has no effect because `-x`  is effectively like specifying files on the command line, and the type limiters have no effect in that case.  Instead, ack should have warned saying "Too bad, that won't work."

(The way to do this was to add `--perl` to the first ack invocation.)

    ack Test::Deep -l --perl | ack -x -L cmp_bag

**Comment by hoelzro** [comment]
@petdance Ditto for `--sort-files`

**Comment by toddr** [comment]
Would it be possible to move where -x is used to allow -x and --perl to work together?

**Comment by pdl** [comment]
I agree that ack should warn about this, because I'm sure I've done this. But the use case is broader and can't always be fixed as follows:

> (The way to do this was to add `--perl` to the first ack invocation.)

This works if the thing you're piping from is ack. But sometimes you have another thing that lists files.

For instance, I would expect to be able to find perl files that are in source control by doing;

    git ls-files | ack -x --perl foo

At the moment, I have to do something like

    git ls-files | ack '\.p[ml]$' | ack -x foo

... which is somewhat more prone to user error.

**Comment by petdance** [comment]
Never mind that `ack '\.p[ml]$'` is not the same as `ack --perl`.

**Comment by petdance** [comment]
Also, not surprisingly, the use of `git ls-files | ack -x --perl foo` is what brought this up.
